jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudformation/CloudFormationTemplate.kt

```diff
@@ -54,6 +54,8 @@ interface NamedMap {
 interface Resource : NamedMap {
     fun isType(requestedType: String): Boolean
     fun type(): String?
+
+    fun getEnvironmentVariables(): Sequence<Variable>
```

Comment:
Environment variables are only on a handful of resources, for our purposes they're available on [`AWS::Lambda::Function`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-function.html) and [`AWS::Serverless::Function`](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction) - both of which are encapsulated in the `software.aws.toolkits.jetbrains.services.cloudformation.Function` implementations - we should probably move this to there.